components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1195,6 +1195,14 @@ func (p *pg) ListRulesForProject(ctx context.Context, projectID string) ([]*v2.R
 		return nil, err
 	}
 
+	// verify project exists, otherwise we just return an empty list
```

Comment:
As soon as you have more than one SQL statement should wrap in a transaction. Granted these are both read operations, but it has a slim chance for unexpected results.